src/frequenz/sdk/timeseries/_formula_engine/_formula_engine.py

```diff
@@ -222,7 +226,7 @@ def __add__(
             A formula builder that can take further expressions, or can be built
                 into a formula engine.
         """
-        return self._higher_order_builder(self) + other  # type: ignore
+        return self._higher_order_builder(self, self._output_type) + other  # type: ignore
```

Comment:
Yup,  it was originally required because the type system is unable to establish a relationship between the 3 generic types in:

```python
class _ComposableFormulaEngine(
    ABC, Generic[_GenericEngine, _GenericHigherOrderBuilder, QuantityT]
):
```

And complains that only one combination is covered in the return statement, not all possible combinations.  I ran into trouble when trying overload methods as well,  don't remember the exact issues.

But anyway, the whole thing has become almost impossible to represent with the super limited Python type system, with the generic quantity types, for reasons mentioned here: https://github.com/frequenz-floss/frequenz-sdk-python/pull/422#discussion_r1226478466

Reply:
OK, maybe it would be good to add a comment somewhere, even if it is vague about why these ignores are needed.

Reply:
done